lib/didComponents.js

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) 2022 Digital Bazaar, Inc. All rights reserved.
+ */
+
+/**
+ * Using a colon (:) as the delimiter, split the identifier
+ * into its components: a scheme, a method, a version, and a
+ * multibaseValue. If there are only three components set the
+ * version to the string value 1 and use the last value as
+ * the multibaseValue.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.did - A did as a string.
+ *
+ * @returns {object} The components of the did.
+*/
+export const getDidKeyComponents = ({did}) => {
+  const {
+    isDidUrl,
+    didEndIndex,
+    fragment,
+    search,
+    searchParams
+  } = _getUrlComponents({did});
+  // the identifier should be the did before the url syntax starts
+  const identifier = isDidUrl ? did.substring(0, didEndIndex) : did;
+  // split on : unless : is the last character in a did url /(?!:$):/
+  const parts = identifier.split(/(?!:$):/);
+  const [scheme, method, version, multibase] = parts;
+  return {
+    scheme,
+    method,
+    // if multibase exists use the version
+    version: multibase ? version : '1',
```

Comment:
This seems like it's wrong, you can have version AND multibase exist at the same time. If the code block is right, it feels confusing. Didn't pass the code smell test... something seems off here.

Reply:
that is exactly what this accounts for `did:key:3:zf00` will result in multibase and version being defined at the same time. If `multibase` is not defined then the string 1 is used as the version.

Reply:
Another problem that needs accounting for here is if someone uses `did:key:1:...` ... that needs to be an error, I'd think.